allennlp/modules/token_embedders/embedding.py

```diff
@@ -88,7 +89,8 @@ def __init__(self,
                  norm_type: float = 2.,
                  scale_grad_by_freq: bool = False,
                  sparse: bool = False,
-                 vocab_namespace: str = None) -> None:
+                 vocab_namespace: str = None,
+                 pretrained_file: str = None) -> None:
```

Comment:
also, are you ever setting this value anywhere?

Reply:
sorry, I meant are any callers of this constructor setting this file? like, is `Embedder.from_params` setting it? it looks like it's not

Reply:
The problem is both are actually `original_pretrained_file`, one is passed and one is (possibly) pre-stored attribute. But, I can distinguish them as `self._initialization_pretrained_file` and `extension_pretrained_file`, where `self._initialization_pretrained_file` will be set at initialization (likely training) and `extension_pretrained_file` will be the argument in `extend_vocab` method. 

If `extension_pretrained_file` is passed and available, it will be used or else we will default to self._initialization_pretrained_file (also look user/stored priority comment).  Sounds good?

Reply:
No, no caller of the constructor is setting the pretrained file except in the constructor itself.

Reply:
I am fine to leave this one `pretrained_file` and have the one in extend called `extension_file` or whatever. Mostly I just don't want there to be a `pretrained_file` and a `self._pretrained_file` that refer to different things, that's super confusing and non-obvious.